src/internal/operators/distinctUntilChanged.ts

```diff
@@ -3,19 +3,38 @@ import { identity } from '../util/identity';
 import { operate } from '../util/lift';
 import { OperatorSubscriber } from './OperatorSubscriber';
 
+export function distinctUntilChanged<T>(comparator?: (previous: T, current: T) => boolean): MonoTypeOperatorFunction<T>;
+export function distinctUntilChanged<T, K>(
+  comparator: (previous: K, current: K) => boolean,
+  keySelector: (value: T) => K
+): MonoTypeOperatorFunction<T>;
+
 /**
  * Returns a result {@link Observable} that emits all values pushed by the source observable if they
  * are distinct in comparison to the last value the result observable emitted.
  *
+ * When provided without parameters or with the first parameter (`{@link distinctUntilChanged#comparator comparator}`),
+ * it behaves like this:
+ *
  * 1. It will always emit the first value from the source.
  * 2. For all subsequent values pushed by the source, they will be compared to the previously emitted values
  *    using the provided `comparator` or an `===` equality check.
  * 3. If the value pushed by the source is determined to be unequal by this check, that value is emitted and
  *    becomes the new "previously emitted value" internally.
  *
+ * When the second parameter (`{@link distinctUntilChanged#keySelector keySelector}`) is provided, the behavior
```

Comment:
The link here doesn't really work for now:

![image](https://user-images.githubusercontent.com/28087049/149145192-2a1f51af-cb0d-4e96-8d39-894223f12dd3.png)

But, I'll try to fix it later. This is how this looks like:

![image](https://user-images.githubusercontent.com/28087049/149145547-09810096-0155-4c9b-ae3d-69ab8dc1d186.png)